prakticni/main.go

```
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+const URL = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+
+type CollectionPerson struct {
+	Person []Person
+}
+
+type Person struct {
+	Name string
+	Age int
+	Passed bool
+	Skills []string
+}
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+func Find(a []string, x string) (int, bool) {
+	for i, n := range a {
+		if x == n {
+			return i, true
+		}
+	}
+	return 0, false
+}
+
+func getPersonsInformation()([]Person, error){
+
+	persons := make([]Person,0)
```

This is created too early, you are starting to use `persons` in line 57, so it should be initialised just above that line.
You actually use it in 2 places before line 57, but in both cases it's just for a return value when an error occurred.

In case when you want to return an error, the first returned value (representing data) is (almost always) not important. In this specific case, you can just return `nil` instead of an empty map.

In either case, it would be good if the method documentation specifies behaviour when an error occurred. You don't yet write documentation, but if you were to write it, you would state "when an error occurs, the first returned value is not defined" and then you can safely return `nil`.